src/providers/twitch/TwitchAccount.cpp

```diff
@@ -301,54 +301,80 @@ void TwitchAccount::loadUserstateEmotes(QStringList emoteSetKeys)
         return;
     }
 
-    getIvr()->getBulkEmoteSets(
-        newEmoteSetKeys.join(","),
-        [this](QJsonArray emoteSetArray) {
-            auto emoteData = this->emotes_.access();
-            for (auto emoteSet : emoteSetArray)
-            {
-                auto newUserEmoteSet = std::make_shared<EmoteSet>();
+    // splitting newEmoteSetKeys to batches of 100, because Ivr API endpoint accepts a maximum of 100 emotesets at once
+    constexpr int batchSize = 100;
 
-                IvrEmoteSet ivrEmoteSet(emoteSet.toObject());
+    std::vector<std::vector<QString>> batches;
 
-                newUserEmoteSet->key = ivrEmoteSet.setId;
+    auto newEmoteSetKeysVector = newEmoteSetKeys.toVector();
+    batches.reserve((newEmoteSetKeysVector.size() + 1) / batchSize);
 
-                auto name = ivrEmoteSet.login;
-                name.detach();
-                name[0] = name[0].toUpper();
+    for (int i = 0; i < newEmoteSetKeysVector.size(); i += batchSize)
+    {
+        auto last = std::min(newEmoteSetKeysVector.size(), i + batchSize);
+        batches.emplace_back(newEmoteSetKeysVector.begin() + i,
+                             newEmoteSetKeysVector.begin() + last);
+    }
 
-                newUserEmoteSet->text = name;
-                newUserEmoteSet->type = QString();
-                newUserEmoteSet->channelName = ivrEmoteSet.login;
+    for (const auto &batch : std::move(batches))
+    {
+        QStringList emoteSetBatch;
+
+        for (const auto &el : batch)
+        {
+            emoteSetBatch.push_back(el);
+        }
 
-                for (const auto &emote : ivrEmoteSet.emotes)
+        getIvr()->getBulkEmoteSets(
+            emoteSetBatch.join(","),
+            [this](QJsonArray emoteSetArray) {
+                auto emoteData = this->emotes_.access();
+                for (auto emoteSet : emoteSetArray)
                 {
-                    IvrEmote ivrEmote(emote.toObject());
+                    auto newUserEmoteSet = std::make_shared<EmoteSet>();
 
-                    auto id = EmoteId{ivrEmote.id};
-                    auto code = EmoteName{ivrEmote.code};
-                    auto cleanCode =
-                        EmoteName{TwitchEmotes::cleanUpEmoteCode(code)};
-                    newUserEmoteSet->emotes.emplace_back(
-                        TwitchEmote{id, cleanCode});
+                    IvrEmoteSet ivrEmoteSet(emoteSet.toObject());
 
-                    emoteData->allEmoteNames.push_back(cleanCode);
+                    newUserEmoteSet->key = ivrEmoteSet.setId;
 
-                    auto twitchEmote =
-                        getApp()->emotes->twitch.getOrCreateEmote(id, code);
-                    emoteData->emotes.emplace(code, twitchEmote);
+                    auto name = ivrEmoteSet.login;
+                    name.detach();
```

Comment:
Why is this needed?

Reply:
Copy-pasta from `TwitchAccount::loadEmoteSetData`, it wasn't actually introduced here, but in #2550 - this is just indentation change - however if this should be changed I could as well make a change in this PR.